ext/vscode/src/commands/installCli.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { DialogResponses, IActionContext } from '@microsoft/vscode-azext-utils';
+import * as os from 'os';
+import { localize } from '../localize';
+import { executeInTerminal } from '../utils/executeInTerminal';
+import { getAzDevTerminalTitle } from './cmdUtil';
+
+const WindowsTerminalCommand = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"`;
+const LinuxTerminalCommand = `curl -fsSL https://aka.ms/install-azd.sh | bash`;
+const MacTerminalCommand = LinuxTerminalCommand; // Same as Linux
+
+export async function installCli(context: IActionContext, shouldPrompt: boolean = true): Promise<void> {
+    if (shouldPrompt) {
+        const message = localize('azure-dev.commands.cli.install.prompt', 'This will install or update the Azure Developer CLI. Do you want to continue?');
+        // Don't need to check the result, if the user chooses cancel a UserCancelledError will be thrown
+        await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, DialogResponses.cancel);
+    }
+
+    let terminalCommand: string;
+
+    if (os.platform() === 'win32') {
+        terminalCommand = WindowsTerminalCommand;
+    } else if (os.platform() === 'linux') {
+        terminalCommand = LinuxTerminalCommand;
+    } else if (os.platform() === 'darwin') {
+        terminalCommand = MacTerminalCommand;
+    } else {
+        context.errorHandling.suppressReportIssue = true;
+        throw new Error(localize('azure-dev.commands.cli.install.unsupportedPlatform', 'Unsupported platform: {0}', os.platform()));
+    }
+
+    // The installation process will be started but not itself awaited, so we won't know the ultimate result
+    await executeInTerminal(terminalCommand, { name: getAzDevTerminalTitle() });

```

Comment:
An unpleasant side effect that I'm not entirely sure how to deal with--the `PATH` environment variable isn't updated in-proc, so a restart of VSCode is needed to start detecting that AZD is installed.

Reply:
What places can it end up on Windows? If it's, e.g., 99% of the time in a single place, I'm fine hard coding that same as *nix.

Reply:
I made a change to stuff the `PATH` variable if we don't see `Azure Dev CLI` in it. This will work most of the time, and covers the case of installing AZD via the VSCode command. It's not as accurate as querying the registry, but I'm honestly reluctant to do that--there are only a couple of Node packages to do that, and they all seemed to be either extremely old or have extremely old dependencies, which makes them generally a CG alert factory.

Reply:
Talked to @savannahostrowski offline; she felt that just _assuming_ the install is to the default `%LOCALAPPDATA%` location is good enough. Since we're doing the install, and using the default arguments, it _should_ always end up there. Less effort than querying the registry (either by probably-deprecated Node packages or by PowerShell).